doc/codebase_anatomy.md

```diff
@@ -0,0 +1,53 @@
+# Codebase Anatomy
+
+This document describes the the file structure of the `kb_sdk` codebase.
+
+#### Root level
+
+* `/bin` - holds the executable files generated on compilation (most importantly, `kb-sdk`).
+* `/doc` - additional documentation about this codebase
+* `Dockerfile` - The dockerfile for the container that runs the SDK
+* `entrypoint` - the entrypoint bash script that is run for the SDK docker container
+* `/img` - a collection of image templates for APP icons
+* `JAR_DEPS` and `JAR_DEPS_BIN` - indexes of java jar dependencies
+* `KBaseJobService.spec` - KIDL spec for a JSON RPC job service
+* `Makefile` - commands for compiling, building docker images, and initializing submodules
+* `Pipfile` and `Pipfile.lock` - python dependencies for pipenv
+* `reports` - Files generated for JaCoCo test coverage reports.
+* `sdkbase` - Docker files for the image used inside actual SDK apps
+* `lib` - Some generic libraries that get included in SDK apps
+* `src` - The main source code for this project. See below.
+* `submodules` and `submodules_hacks` - jars and other dependencies
+
+#### Source code in `/src/java/us/kbase`
+
+* `/catalog` - Services for managing, registering, and building modules in the catalog. All code here is duplicated in the `njs_wrapper`.
+* `/common/executionengine` - Code for executing jobs and sub-jobs. All the code here is duplicated in the `njs_wrapper` repo
+* `/common/service` - Some tuple datatypes
+* `/common/utils` - NetUtils for working with IP addresses and ports
+* `/jkidl` - Functionality for parsing KIDL spec files
+* `/kidl` - KIDL parser syntax types
+* `/kbasejobservice` - Utils for handling RPC jobs
+* `/mobu` - Module Builder (see below)
+* `/narrativemethodstore` - JSON specifications for SDK app configurations (eg spec.json)
```

Comment:
Same here, this is a compiled client

Reply:
"Compiled client" isn't a phrase I'm familiar with, and it's hard to tell what that means -- ie. "compiled" from what to what and what is a "client" of what server?

Maybe it would be more descriptive to document these as Java classes that map to json/yaml data structures using Jackson annotations. But I'm not sure where all this json is used -- some of it looks like spec.json but also it seems like there's a lot more going on

Reply:
In this case it's compiled from the narrative method store KIDL spec and is a client for the narrative method store server. Same deal for the catalog for the previous comment.